woodwork/utils.py

```diff
@@ -515,6 +515,20 @@ def concat_columns(objs, validate_schema=True):
 
     combined_df = lib.concat(objs, axis=1, join="outer")
 
+    # The lib.concat breaks the woodwork schema for dataframes with different shapes
+    # or mismatched indices.
+    null_cols = combined_df.isnull().any()[combined_df.isnull().any()].index
+    if "dask" not in lib.__name__:
+        null_cols = null_cols.to_numpy()
+    else:
+        null_cols = list(null_cols)
+    for null_col in null_cols:
+        if null_col in logical_types and isinstance(
+            logical_types[null_col],
+            ww.logical_types.Integer,
```

Comment:
I expect this needs to be extended to Boolean/BooleanNullable

Reply:
We don't want to include that here?

Reply:
We can't yet because of https://github.com/alteryx/woodwork/issues/1486

Reply:
Working on that now